packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -174,9 +174,11 @@ export class Publisher {
       return ctxPromise
     } else {
       // this should only occur if max event size is exceeded
-      ctx.setFailedDelivery({
-        reason: new Error(fbAddStatus.message),
-      })
+      if (!fbAddStatus.success) {
+        ctx.setFailedDelivery({
```

Comment:
This diff gives me pause. If we need this to satisfy the type checker, I'd rather not make this change =)

Reply:
I thought adding a type for the different return types would be useful, but we can also resolve this error by using:

```
      if ('message' in fbAddStatus) {
        ctx.setFailedDelivery({
          reason: new Error(fbAddStatus.message),
        })
      }
```